public/app/core/components/TimeSeries/utils.ts

```diff
@@ -553,6 +555,69 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn = ({
 
   builder.setStackingGroups(stackingGroups);
 
+  const shouldShowValues = frame.fields.slice(1).some((field) => {
+    const customConfig = { ...defaultConfig, ...field.config.custom };
+    if (customConfig.showValues === VisibilityMode.Always) {
+      return true;
+    }
+    if (
+      customConfig.showValues === VisibilityMode.Auto &&
+      field.type === FieldType.number &&
+      customConfig.showPoints !== VisibilityMode.Never
+    ) {
+      return true;
+    }
+    return false;
+  });
+
+  if (shouldShowValues) {
+    builder.addHook('draw', (u: uPlot) => {
+      const { ctx } = u;
+      ctx.save();
+      ctx.fillStyle = theme.colors.text.primary;
+      ctx.font = `12px ${theme.typography.fontFamily}`;
+      ctx.textAlign = 'center';
+      ctx.textBaseline = 'bottom';
+
+      // Iterate through series (skip x-axis series at index 0)
+      for (let seriesIdx = 1; seriesIdx < u.data.length; seriesIdx++) {
+        const field = frame.fields[seriesIdx];
+        if (!field) {
+          continue;
+        }
```

Comment:
by the time we get to this "low level" part of the code, i dont think this is possible. we dont have these checks in other places.